pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -3015,6 +3015,39 @@ func TestExternalDNSDeploymentChanged(t *testing.T) {
 				}),
 			}),
 		},
+		{
+			description: "if externalDNS has extra volume",
+			expect:      false,
+			originalDeployment: testDeploymentWithVolumes(
+				testSecretVolume("testcreds", "testsecret", "creds", "/run/secrets"),
+				testConfigMapVolume("extravolume", "extra", "files", "/etc"),
+			),
+			mutate: func(dep *appsv1.Deployment) {
+				dep.Spec.Template.Spec.Volumes = []corev1.Volume{testSecretVolume("testcreds", "testsecret", "creds", "/run/secrets")}
+			},
+			expectedDeployment: testDeploymentWithVolumes(
+				testSecretVolume("testcreds", "testsecret", "creds", "/run/secrets"),
+				testConfigMapVolume("extravolume", "extra", "files", "/etc"),
+			),
+		},
+		{
+			description:        "if ignored field was added to secret volume",
+			expect:             false,
+			originalDeployment: testDeploymentWithVolumes(testSecretVolume("testcreds", "testsecret", "creds", "/run/secrets")),
+			mutate: func(dep *appsv1.Deployment) {
+				dep.Spec.Template.Spec.Volumes[0].VolumeSource.Secret.DefaultMode = nil
+			},
+			expectedDeployment: testDeploymentWithVolumes(testSecretVolume("testcreds", "testsecret", "creds", "/run/secrets")),
+		},
+		{
+			description:        "if ignored field was added to configmap volume",
+			expect:             false,
+			originalDeployment: testDeploymentWithVolumes(testConfigMapVolume("testcerts", "testcerts", "key", "/etc/pki/trust")),
+			mutate: func(dep *appsv1.Deployment) {
+				dep.Spec.Template.Spec.Volumes[0].VolumeSource.ConfigMap.DefaultMode = nil
```

Comment:
```suggestion
				dep.Spec.Template.Spec.Volumes[0].VolumeSource.ConfigMap.DefaultMode = somethingnotnil
```

Reply:
`nil` seems correct to me.  Maybe the test case would be clearer if it were named something like "if the API sets a default mode on a configmap volume".

Reply:
Changed the name to the suggested.